### PR TITLE
Creator Frontend + Backend Changes and Stitching

### PR DESCRIPTION
**Pull request by NathanSouphanthong**
## Notion ticket link

[Creator Frontend + Backend Changes and stitching](https://www.notion.so/uwblueprintexecs/Creator-Frontend-Backend-Changes-and-stitching-49a45413e4234ff69f8a0f31eb9b4774)

## Implementation description
* Added creator types (Creator, Presentation, Publication, CreatorRequest) in frontend/src/types/CreatorTypes 
* Added updateCreator method in backend + exposed it in route /creators/<id>
* Modified createCreator method to take in only userId
* Added CreatorAPIClient.ts for updateCreator and createCreator functions

## What should reviewers focus on?
* Updates to createCreator method on backend
* CreatorAPIClient updateCreator, createCreator functions

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have run docker-compose up and my project compiled
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/APIClients/CreatorAPIClient.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  try {
+    const { data } = await baseAPIClient.post("/", newCreator, {
+      headers: {
+        Authorization: getBearerToken(),
+        "Content-Type": "application/json",
+      },
+    });
+    return data;
+  } catch (error) {
+    return null;
```
Comment (Shrinjay) [inline frontend/src/APIClients/CreatorAPIClient.ts:18]: Add some logging for the error case, can be as simple as a console.log.

File: frontend/src/APIClients/CreatorAPIClient.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  try {
+    const { data } = await baseAPIClient.put(`/creators/${userId}`, creator, {
+      headers: {
+        Authorization: getBearerToken(),
+        "Content-Type": "application/json",
+      },
+    });
+    return data;
+  } catch (error) {
+    return null;
```
Comment (Shrinjay) [inline frontend/src/APIClients/CreatorAPIClient.ts:35]: Same here, add logging

**Review by Shrinjay — CHANGES_REQUESTED** [review]
Looking good man, just add logging

File: backend/typescript/services/implementations/creatorService.ts [hunk 3]
```diff
@@ -150,6 +150,33 @@ class CreatorService implements ICreatorService {
...
+        {
+          location: creator.location,
+          rate: creator.rate,
+          genre: creator.genre,
+          ageRange: creator.ageRange,
+          timezone: creator.timezone,
+          bio: creator.bio,
+        },
+        {
+          where: { userId },
```
Comment (NathanSouphanthong) [inline backend/typescript/services/implementations/creatorService.ts:168]: Missed this, should be `where: {userId: userId}`

**Comment by Shrinjay** [comment]
@NathanSouphanthong [video going over backend testing with postman](https://drive.google.com/file/d/1xsFyp_EMogtXipR-AXgiNRAljy20P8h7/view?usp=sharing)

**Review by Shrinjay — APPROVED** [review]

**Pull request merged**